include/language/en_us.lang.php

```diff
@@ -42,7 +42,7 @@
     die('Not A Valid Entry Point');
 }
 
-include_once __DIR__ . '/../SugarPHPMailer.php';
++include_once __DIR__ . '/../SugarEmailAddress/SugarEmailAddress.php';
```

Comment:
Will this be backward compatible (the 7.10 language file can be used by oldest SuiteCRM versions?)

Also I consider that including code files in a language files is looking for problems ahead.
If there is a parsing error in some code file it can cause the language file not to load and break the application.
There should be another way! 
I think the language files should be cleaner, slim and neutral on code to reduce errors.

Reply:
I agree that the language files should be as clean as possible

Both SugarPHPMailer and SugarEmailAddress have existed since SuiteCRM 7.1. The number of references to SugarEmailAddress and SugarPHPMailer in the languages files, will determine if this is upgrade safe. As far as I can tell, there are 0 references to SugarPHPMailer. So I believe this is correct. It should not cause any issues.

Reply:
@horus68  - @daniel-samson is looking at this currently and reverting the include.

Reply:
I will remove the include completely and then remove the references to SugarEmailAddress. This should keep it backwards compatible.

However, there is a bigger discussion to be had, regarding how long we support language packs, and languages packs should contain code inside them.

Reply:
My say on this:
- Language pack support: when backward compatibility is lost we will have a locked language pack for old versions (a zip repo) and maintain/update only the new version. Crowdin can have branch (versions) for language files but I will not use that because then we will lost the auto-package feature from Crowdin (the actual option).

- Languages packs containing code inside them: For me is a big NO. Language files are translators area. Code needs to be packed with core code files. Translations should not be a potential error source, in form of including code files or placing code inside some strings.
Actually there are some code inside language strings and files like the include/ language file causes many errors for parsing code and making the interface not usable for some languages.
Its also too difficult to find errors in language files on languages with different alphabets. 
So less code, less time consuming finding errors errors.

Reply:
Forum topic for Language pack support
https://suitecrm.com/suitecrm/forum/international-language-support/17639-language-packs-backward-compatibility-discussion

Reply:
OK. Well, in order to remove the code from the languages files. We will need to move the SugarThemeRegistry stuff out into php, smarty templates or JavaScript files. 

What about the labels which contain SQL and HTML?

Reply:
If HTML and SQL will be out of languages files even better.

Reply:
I agree with @horus68  about excluding code from language files, but HTML is sometimes used in Russian langpack to highlight text in bold or provide a link to some resources, such as a user's guide for russian users, etc..
 I agree that this is not entirely correct, but sometimes it is convenient to pay user's attention to certain nuances, and in different language packs these nuances may vary. It would be nice if the translator had more tools for a quality translation.

Reply:
It is worth mentioning, that we do not support all versions of SuiteCRM. Each version has an end of life. Should the language packs be different?

How far back should the language packs be supported?

Reply:
I think it will be better to work with a single universal package that will work in all supported versions  
(after (1) removing the code from the language files  and  (2) after  patching the corresponding supported SuiteCRM versions) .